core/unix/os.c

```diff
@@ -3719,9 +3719,11 @@ dr_create_client_thread(void (*func)(void *param), void *arg)
 #        endif
     LOG(THREAD, LOG_ALL, 1, "dr_create_client_thread xsp=" PFX " dstack=" PFX "\n", xsp,
         get_clone_record_dstack(crec));
+    os_clone_pre(dcontext);
```

Comment:
Should we be worried that os_clone_pre, which is meant to handle an app thread doing a clone, will in some future change do something extra that we do not want done for cloning while in DR mode?  At least a comment should be added to os_clone_{pre,post} that they are used for client threads too and they should not do anything that assumes the app.

Reply:
Ok.

Reply:
Done.